src/onnxruntime.cc

```diff
@@ -388,221 +388,219 @@ ModelState::LoadModel(
   // to indicate that execution providers should not be added (this is
   // just a local convention to this function, not the standard
   // interpretation of AUTO).
-  if (instance_group_kind != TRITONSERVER_INSTANCEGROUPKIND_AUTO) {
-    // Don't need to ensure uniqueness of the providers, ONNX Runtime
-    // will check it.
+  // Don't need to ensure uniqueness of the providers, ONNX Runtime
+  // will check it.
 
-    // GPU execution providers
+  // GPU execution providers
 #ifdef TRITON_ENABLE_GPU
-    if (instance_group_kind == TRITONSERVER_INSTANCEGROUPKIND_GPU) {
-      triton::common::TritonJson::Value optimization;
-      if (model_config_.Find("optimization", &optimization)) {
-        triton::common::TritonJson::Value eas;
-        if (optimization.Find("execution_accelerators", &eas)) {
-          triton::common::TritonJson::Value gpu_eas;
-          if (eas.Find("gpu_execution_accelerator", &gpu_eas)) {
-            for (size_t ea_idx = 0; ea_idx < gpu_eas.ArraySize(); ea_idx++) {
-              triton::common::TritonJson::Value ea;
-              RETURN_IF_ERROR(gpu_eas.IndexAsObject(ea_idx, &ea));
-              std::string name;
-              RETURN_IF_ERROR(ea.MemberAsString("name", &name));
+  if ((instance_group_kind == TRITONSERVER_INSTANCEGROUPKIND_GPU) &&
```

Comment:
shouldn't this be:
`(instance_group_kind == TRITONSERVER_INSTANCEGROUPKIND_GPU) || (instance_group_kind == TRITONSERVER_INSTANCEGROUPKIND_AUTO)`
?
Even with kind_auto, the config may specify TensorRT execution provider optimization.

Reply:
We do not want to add execution accelerators when it is TRITONSERVER_INSTANCEGROUPKIND_AUTO right? I may have misunderstood

Reply:
TRITONSERVER_INSTANCEGROUPKIND_AUTO should be treated exactly like TRITONSERVER_INSTANCEGROUPKIND_GPU when adding the Execution Providers.

Assume a user providing the following config:

```
instance_group [ { kind: KIND_AUTO }]

optimization { execution_accelerators {
  gpu_execution_accelerator : [ {
    name : "tensorrt"
    parameters { key: "precision_mode" value: "FP16" }
    parameters { key: "max_workspace_size_bytes" value: "1073741824" }
    }]
}}
``` 
We need to add the TensorRT EP in this case.